### PR TITLE
fix(consent-engine): correct GORM tag bugs breaking the active-consent unique index

### PR DESCRIPTION
**Pull request by sthanikan2000**
## Summary

Fixes a blocker reported against a fresh Consent Engine database: `RUN_MIGRATION=true` fails GORM's `AutoMigrate` with `ERROR: syntax error at end of input (SQLSTATE 42601)` because the `idx_consent_active_unique` index tag on `ConsentRecord` embeds a comma inside `where:status IN ('pending', 'approved')`, which GORM's tag parser splits on, truncating the WHERE clause. The same tag also uses `index:` instead of `uniqueIndex:`, so even with the syntax fixed, the resulting index would not enforce the "only one pending/approved consent per (owner_id, owner_email, app_id)" business rule.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- `exchange/consent-engine/v1/models/consent.go`: on the `OwnerID`, `OwnerEmail`, `AppID`, and `Status` fields, changed `index:idx_consent_active_unique` → `uniqueIndex:idx_consent_active_unique` and rewrote `where:status IN ('pending', 'approved')` → `where:status = 'pending' OR status = 'approved'` (comma-free, semantically identical predicate).
- No other files changed — AutoMigrate/`RUN_MIGRATION` behavior is otherwise unchanged.

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

Verified by reading GORM's vendored tag-parsing source (`schema/index.go`, `schema/utils.go`) to confirm the exact root cause, then reproduced the original failure end-to-end: built the binary and ran it with `RUN_MIGRATION=true` against a genuinely fresh `postgres:15-alpine` container. Before this fix that scenario fails; after it, `AutoMigrate` completes successfully and `\d consent_records` shows `idx_consent_active_unique` as a true `UNIQUE` index with the correct predicate:
```
"idx_consent_active_unique" UNIQUE, btree (owner_id, owner_email, app_id, status) WHERE status::text = 'pending'::text OR status::text = 'approved'::text
```
Existing `v1/services` and `v1/handlers` test suites pass unchanged.

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

N/A

## Additional Notes

Scoped intentionally to just the two tag bugs — no change to how/when migration runs (`RUN_MIGRATION`-gated `AutoMigrate` stays as-is). This is a no-op for any environment where `idx_consent_active_unique` already exists correctly (e.g. via the existing hand-written `add_consent_unique_tuple_constraint.sql`), since GORM's migrator skips creating an index that already exists by name.

## Deployment Notes

None — no schema migration required for this change itself; it only affects what GORM does the next time `AutoMigrate` runs against a database that doesn't yet have `idx_consent_active_unique`.

### CODE REVIEW

File: exchange/consent-engine/v1/models/consent.go [hunk 1]
```diff
@@ -16,18 +16,18 @@ type ConsentRecord struct {
...
+	AppID string `gorm:"column:app_id;type:varchar(255);not null;index:idx_consent_records_app_id;index:idx_consent_records_owner_app,composite:owner_app;uniqueIndex:idx_consent_active_unique,composite:active_unique,where:status = 'pending' OR status = 'approved'" json:"app_id"`
 	// AppName is the name of the consumer application
 	AppName *string `gorm:"column:app_name;type:varchar(255);" json:"app_name,omitempty"`
 	// Status is the status of the consent record: pending, approved, rejected, expired, revoked
 	// Part of conditional unique constraint for active consents (pending/approved)
-	Status string `gorm:"column:status;type:varchar(50);not null;index:idx_consent_records_status;index:idx_consent_active_unique,composite:active_unique,where:status IN ('pending', 'approved')" json:"status"`
+	Status string `gorm:"column:status;type:varchar(50);not null;index:idx_consent_records_status;uniqueIndex:idx_consent_active_unique,composite:active_unique,where:status = 'pending' OR status = 'approved'" json:"status"`
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply exchange/consent-engine/v1/models/consent.go]: Good catch, but this is a pre-existing gap rather than something introduced by this PR — the same column composition `(owner_id, owner_email, app_id, status)` for `idx_consent_active_unique` is already what's in `exchange/consent-engine/v1/migrations/add_consent_unique_tuple_constraint.sql`, which predates this change and is what most environments' schema is actually provisioned from today (this PR's tag fix only affects fresh databases where AutoMigrate creates the index from scratch).

This PR is intentionally scoped to just the two bugs that caused the reported blocker (the comma-truncated `where:` clause and the missing `unique` semantics), per LSFLK/lsf-govtech-ndx#1. Properly fixing the tuple composition would mean changing actual constraint semantics and updating the SQL migration + integration test SQL to match, plus a real migration for already-provisioned databases — a larger, separate change. I'll track this as a follow-up rather than rolling it into this fix.

File: exchange/consent-engine/v1/models/consent.go [hunk 2]
```diff
@@ -16,18 +16,18 @@ type ConsentRecord struct {
 	ConsentID uuid.UUID `gorm:"column:consent_id;type:uuid;primaryKey;default:gen_random_uuid()" json:"consent_id"`
 	// OwnerID is the unique identifier for the data owner
 	// Part of conditional unique constraint for active consents (pending/approved)
-	OwnerID string `gorm:"column:owner_id;type:varchar(255);not null;index:idx_consent_records_owner_id;index:idx_consent_records_owner_app,composite:owner_app;index:idx_consent_active_unique,composite:active_unique,where:status IN ('pending', 'approved')" json:"owner_id"`
+	OwnerID string `gorm:"column:owner_id;type:varchar(255);not null;index:idx_consent_records_owner_id;index:idx_consent_records_owner_app,composite:owner_app;uniqueIndex:idx_consent_active_unique,composite:active_unique,where:status = 'pending' OR status = 'approved'" json:"owner_id"`
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply exchange/consent-engine/v1/models/consent.go]: Fixed in 27ebbafb — removed the redundant `composite:` fragments.

File: exchange/consent-engine/v1/models/consent.go [hunk 3]
```diff
@@ -16,18 +16,18 @@ type ConsentRecord struct {
...
 	// OwnerEmail is the email address of the data owner
 	// Part of conditional unique constraint for active consents (pending/approved)
-	OwnerEmail string `gorm:"column:owner_email;type:varchar(255);not null;index:idx_consent_records_owner_email;index:idx_consent_active_unique,composite:active_unique,where:status IN ('pending', 'approved')" json:"owner_email"`
+	OwnerEmail string `gorm:"column:owner_email;type:varchar(255);not null;index:idx_consent_records_owner_email;uniqueIndex:idx_consent_active_unique,composite:active_unique,where:status = 'pending' OR status = 'approved'" json:"owner_email"`
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply exchange/consent-engine/v1/models/consent.go]: Fixed in 27ebbafb — removed the redundant `composite:` fragments.

File: exchange/consent-engine/v1/models/consent.go [hunk 4]
```diff
@@ -16,18 +16,18 @@ type ConsentRecord struct {
...
 	// AppID is the unique identifier for the consumer application
 	// Part of conditional unique constraint for active consents (pending/approved)
-	AppID string `gorm:"column:app_id;type:varchar(255);not null;index:idx_consent_records_app_id;index:idx_consent_records_owner_app,composite:owner_app;index:idx_consent_active_unique,composite:active_unique,where:status IN ('pending', 'approved')" json:"app_id"`
+	AppID string `gorm:"column:app_id;type:varchar(255);not null;index:idx_consent_records_app_id;index:idx_consent_records_owner_app,composite:owner_app;uniqueIndex:idx_consent_active_unique,composite:active_unique,where:status = 'pending' OR status = 'approved'" json:"app_id"`
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply exchange/consent-engine/v1/models/consent.go]: Fixed in 27ebbafb — removed the redundant `composite:` fragments.

**Review by mushrafmim — APPROVED** [review]
